tips/tests/test_rule_engine.py

```diff
@@ -239,3 +104,268 @@ def test_apply_rules_recursing(self):
+        }
+        rules = [
+            {"type": "rule", 
+            "rule": "len($.brp.kinderen[now() - timeDelta(10, 0, 0, 0, 0, 0) >= dateTime(@.geboortedatum) and now() - timeDelta(13, 0, 0, 0, 0, 0) < dateTime(@.geboortedatum)]) >= 1"}
+        ]
+        self.assertFalse(apply_rules(user_data, rules, rule))
+
+        fixture["data"]['brp']['kinderen'][0]['geboortedatum'] = '2010-01-01T00:00:00Z'
+        fixture["data"]['brp']['kinderen'][1]['geboortedatum'] = '2010-01-01T00:00:00Z'
+        user_data = objectpath.Tree(fixture["data"])
+        self.assertTrue(apply_rules(user_data, rules, rule))
+
+        fixture["data"]['brp']['kinderen'][0]['geboortedatum'] = '2009-01-01T00:00:00Z'
+        fixture["data"]['brp']['kinderen'][1]['geboortedatum'] = '2009-01-01T00:00:00Z'
+        user_data = objectpath.Tree(fixture["data"])
+        self.assertTrue(apply_rules(user_data, rules, rule))
+
+        fixture["data"]['brp']['kinderen'][0]['geboortedatum'] = '2008-01-01T00:00:00Z'
+        fixture["data"]['brp']['kinderen'][1]['geboortedatum'] = '2008-01-01T00:00:00Z'
+        user_data = objectpath.Tree(fixture["data"])
+        self.assertTrue(apply_rules(user_data, rules, rule))
+
+        fixture["data"]['brp']['kinderen'][0]['geboortedatum'] = '2007-01-01T00:00:00Z'
+        fixture["data"]['brp']['kinderen'][1]['geboortedatum'] = '2007-01-01T00:00:00Z'
+        user_data = objectpath.Tree(fixture["data"])
+        self.assertFalse(apply_rules(user_data, rules, rule))
+
+        fixture["data"]['brp']['kinderen'][0]['geboortedatum'] = '20011-01-01T00:00:00Z'
+        fixture["data"]['brp']['kinderen'][1]['geboortedatum'] = '20011-01-01T00:00:00Z'
+        user_data = objectpath.Tree(fixture["data"])
+        self.assertFalse(apply_rules(user_data, rules, rule))
+
+    def test_is_66_of_ouder(self):
+        fixture = get_fixture()
+        user_data = objectpath.Tree(fixture["data"])
+        rule = {
+                "1": {
+                "name": "is 66",
+                "rules": [
+                    {"type": "rule", 
+                    "rule": "dateTime($.brp.persoon.geboortedatum) + timeDelta(66, 4, 0, 0, 0, 0) <= now()"}
+                ]
+            }
+        }
+        rules = [
+            {"type": "rule", 
+            "rule": "dateTime($.brp.persoon.geboortedatum) + timeDelta(66, 4, 0, 0, 0, 0) <= now()"}
+        ]
+        self.assertTrue(apply_rules(user_data, rules, rule))
+
+        fixture["data"]['brp']['persoon']['geboortedatum'] = '1950-01-01T00:00:00Z'
+        user_data = objectpath.Tree(fixture["data"])
+        self.assertTrue(apply_rules(user_data, rules, rule))
+
+        fixture["data"]['brp']['persoon']['geboortedatum'] = '2000-01-01T00:00:00Z'
+        user_data = objectpath.Tree(fixture["data"])
+        self.assertFalse(apply_rules(user_data, rules, rule))
+
+    def test_nationaliteit(self):
+        fixture = get_fixture()
+        user_data = objectpath.Tree(fixture["data"])
+        rule = {
```

Comment:
Is dit een soort compound rules? de vars `rule` en `rules` lijken iets teveel op elkaar.